src/viam/resource/registry.py

```diff
@@ -70,8 +74,9 @@ class Registry:
     resource using ``Registry.register(...)``.
     """
 
-    _SUBTYPES: ClassVar[Dict[Subtype, ResourceRegistration]] = {}
-    _COMPONENTS: ClassVar[Dict[str, ComponentCreator]] = {}
+    _SUBTYPES: ClassVar[Dict["Subtype", ResourceRegistration]] = {}
+    _COMPONENTS: ClassVar[Dict[str, "ComponentCreator"]] = {}
```

Comment:
I do feel fairly strongly about only having one Creator registry (even if we have to do a `Union["ComponentCreator","ServiceCreator"]` value typing) - I think it'll make future extensions easier. thoughts?

Reply:
That was the original plan, but we decided to follow [RDK's lead](https://github.com/viamrobotics/rdk/blob/c720c2e49cac884fb0ad1c4c0bb0205256fa900b/registry/registry.go#L161). Happy to change it but I also don't think it limits us too much. These are internal types, we can always rename/rearrange them later without breaking.

I also think this might do the opposite of limit us. Any future extensions might not conform nicely to either of those types, and then we'd just keep growing the types allowed. By keeping these separate, we can add new types as we see fit without changing existing data structures.

Reply:
the longer term goal for the RDK is to also move to a single registry as well. I think growing the types allowed is fine, as long as the keys can be uniquely generated. One additional feature we're adding is config validation, and adding it in one place is a bit nicer than two. We might add more types later, but I would push for making the function signatures be the same (takes a config and returns some kind of resource). 
I agree about the bit that it is mostly internal, but the external bit (how devs register new models) would change if we combine the registries later on. So in that sense, the ask is actually to allow devs to register new component/service models the same way.